File: 4107-find-missing-elements/4107-find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        n = len(nums)
        maximum, minimum = -inf, inf
        vis = [False] * 101
        
        for i in range(n):
            maximum = max(maximum, nums[i])
            minimum = min(minimum, nums[i])
            vis[nums[i]] = True

        res = []
        for i in range(minimum, maximum):
            if not vis[i]:
                res.append(i)

        return res
        

File: 4107-find-missing-elements/4107-find-missing-elements.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* findMissingElements(int* nums, int numsSize, int* returnSize) {
    int* ans = (int*)calloc(sizeof(int), 101);
    int min = INT_MAX;
    int max = INT_MIN;
    bool vis[101] = {};
    *returnSize = 0;
    for (int x = 0; x < numsSize; ++x) {
        min = fmin(min, nums[x]);
        max = fmax(max, nums[x]);
        vis[nums[x]] = true;
    }

    for (int i = min; i < max; ++i) {
        if (!vis[i])
            ans[(*returnSize)++] = i;
    }
    return ans;
}